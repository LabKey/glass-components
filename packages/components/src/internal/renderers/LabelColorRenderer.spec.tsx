import React from 'react';
import { fromJS } from 'immutable';
import renderer from 'react-test-renderer';

import { LabelColorRenderer } from './LabelColorRenderer';

describe('LabelColorRenderer', () => {
    test('undefined data', () => {
        const component = <LabelColorRenderer data={undefined} />;
        const tree = renderer.create(component).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('empty data', () => {
        const component = <LabelColorRenderer data={fromJS({})} />;
        const tree = renderer.create(component).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('with data', () => {
        const component = <LabelColorRenderer data={fromJS({ value: '#000000' })} />;
        const tree = renderer.create(component).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
