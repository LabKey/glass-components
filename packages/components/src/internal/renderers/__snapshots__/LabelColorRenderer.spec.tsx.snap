// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelColorRenderer empty data 1`] = `null`;

exports[`LabelColorRenderer undefined data 1`] = `null`;

exports[`LabelColorRenderer with data 1`] = `
<i
  className="color-icon__circle"
  style={
    {
      "backgroundColor": "#000000",
    }
  }
/>
`;
